lib/cc/yaml/nodes/glob_list.rb

```diff
@@ -3,6 +3,10 @@ module Yaml
     module Nodes
       class GlobList < Sequence
         type Glob
+
+        def allow_child?(child)
```

Comment:
I stand by my previous opinion:

> this workaround will allow us to be
stricter in cases where it matters, without changing things more broadly which
has unknown risk.

For example, would this code be be hit by the "arbitrary" `engines.x.config` nodes?

If you don't share those concerns, I can move this directly into `Sequence` -- WDYT?